whisperv6/filter.go

```diff
@@ -26,6 +26,46 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
+// MessageStore defines interface for temporary message store.
+type MessageStore interface {
```

Comment:
It sounds more like `MessageStash`, like `git stash`, and I would name these method `Save` and `Pop` or something like that.

Then the behaviour (clean on retrieve) will be expected.